runtime/src/bank.rs

```diff
@@ -161,6 +161,11 @@ pub struct TransactionResults {
     pub fee_collection_results: Vec<Result<()>>,
     pub processing_results: Vec<TransactionProcessResult>,
 }
+pub struct TransactionBalanceSet {
```

Comment:
how about:
```suggestion
pub struct TransactionBalancesSet {
```
Until I realized that `TransactionBalances` is a vector of vectors, I assumed `TransactionBalanceSet` was just for a single transaction

Reply:
I'm not a big fan of the implicit assumption that the pre and post balance vectors must be of equal length, but 🙉.  I appreciate how doing so simplifies other parts of the implementation

Reply:
I could add an assert on the length equality when TransactionBalancesSet is built? `collect_balances` operates on the same immutable transaction list for pre and post, so it should always be true.

Reply:
Throwing in an `assert` somewhere would be great, it might save somebody hours debugging one day